flask_app/tasks.py

```diff
@@ -111,30 +129,35 @@ def create_key(s: str) -> str:
 Montgomery Scott
 """
 
-def _get_combadge_by_version(combadge_version: str) -> bytes:
-    return {
-        "v1": _COMBADGE_V1,
-        "v2": _COMBADGE_V2,
-    }[combadge_version]
-
-
-def _upload_combadge(ssh_client: SSHClient, combadge_version: str) -> str:
-    _, stdout, stderr = ssh_client.exec_command("mktemp /tmp/combadge.XXX")
+def _get_os_type(ssh_client: SSHClient) -> str:
+    _, stdout, stderr = ssh_client.exec_command("uname")
     retcode = stdout.channel.recv_exit_status()
     if retcode != 0:
-        raise Exception(stderr.read().decode("utf-8"))
+        return 'windows'
+    return stdout.read().decode("utf-8").strip().lower()
 
-    combadge_path = stdout.read().decode("utf-8").strip()
 
-    stdin, stdout, stderr = ssh_client.exec_command(f"sh -c \"cat > {combadge_path} && chmod +x {combadge_path}\"")
-    combadge = _get_combadge_by_version(combadge_version)
-    stdin.write(combadge)
-    stdin.channel.shutdown_write()
-    retcode = stdout.channel.recv_exit_status()
-    if retcode != 0:
-        raise Exception(stderr.read().decode("utf-8"))
+def _generate_random_combadge_name(stringLength: int) -> str:
```

Comment:
I think it's safer to use `mktemp` when it's available as it guarantees the file doesn't exist on the host.

Reply:
At this point we agreed to use the existing version without `mktemp`